app/controllers/pets_controller.rb

```diff
@@ -33,6 +34,29 @@ def show
   end
 
   def edit
+    @pet = Pet.find(params[:id])
+    @pet_images = PetForm.new
+  end
+
+  def update
+    @pet = Pet.find(params[:id])
+
+    ActiveRecord::Base.transaction do
+      @pet.update!(pet_params)
+      if params[:pet_form]
+        present?
```

Comment:
↓ですかね？（`.present?` は無くても良いですが）

```suggestion
      if params[:pet_form].present?
```

`pet_images.present?` でも良さそうです。